lmdeploy/serve/async_engine.py

```diff
@@ -30,51 +39,107 @@ class AsyncEngine:
                     "InternLM/internlm-chat-20b-4bit",
                     "lmdeploy/llama2-chat-70b-4bit", etc.
                 - iii) The model_id of a model hosted inside a model repo
-                    on huggingface.co, such as "InternLM/internlm-chat-7b",
+                    on huggingface.co, such as "internlm/internlm-chat-7b",
                     "Qwen/Qwen-7B-Chat ", "baichuan-inc/Baichuan2-7B-Chat"
                     and so on.
         model_name (str): needed when model_path is a pytorch model on
-            huggingface.co, such as "InternLM/internlm-chat-7b",
+            huggingface.co, such as "internlm/internlm-chat-7b",
             "Qwen/Qwen-7B-Chat ", "baichuan-inc/Baichuan2-7B-Chat" and so on.
+        backend (str): either `turbomind` or `pytorch` backend. Default to
+            `turbomind` backend.
+        backend_config (EngineConfig): beckend config. Default to none.
+        chat_template_config (ChatTemplateConfig): chat template configuration.
+            Default to None.
         instance_num (int): instance numbers to be created
         tp (int): tensor parallel
     """
 
     def __init__(self,
                  model_path: str,
                  model_name: Optional[str] = None,
+                 backend: Literal['turbomind', 'pytorch'] = 'turbomind',
+                 backend_config: Optional[Union[TurbomindEngineConfig,
+                                                PytorchEngineConfig]] = None,
+                 chat_template_config: Optional[ChatTemplateConfig] = None,
                  instance_num: int = 32,
                  tp: int = 1,
                  **kwargs) -> None:
-        from lmdeploy import turbomind as tm
-        self.tm_model = tm.TurboMind.from_pretrained(model_path,
-                                                     model_name=model_name,
-                                                     tp=tp,
-                                                     **kwargs)
-        self.tokenizer = self.tm_model.tokenizer
+        if backend == 'turbomind':
```

Comment:
Advice about code refactoring:
mix branch condition and branch body would make the code hard to read(especially with nested branches and loops). I prefer to pack the branch body into separate functions such as:
```python
if backend == 'turbomind':
    _build_turbomind(...)
elif backend == 'pytorch':
    _build_pytorch(...)
else:
    raise ...
```
All conditions are gathered in the same screen, and the nested body would be shallow.